sdk/keyvault/azsecrets/README.md

```diff
@@ -0,0 +1,272 @@
+az ad sp create-for-rbac --name http://my-application --skip-assignment
+```
+
+> Output:
+> ```json
+> {
+>     "appId": "generated app id",
+>     "displayName": "my-application",
+>     "name": "http://my-application",
+>     "password": "random password",
+>     "tenant": "tenant id"
+> }
+> ```
+
+Use the output to set **AZURE_CLIENT_ID** ("appId" above), **AZURE_CLIENT_SECRET** ("password" above) and **AZURE_TENANT_ID** ("tenant" above) environment variables. The following example shows a way to do this in Bash:
+```Bash
+export AZURE_CLIENT_ID="generated app id"
+export AZURE_CLIENT_SECRET="random password"
+export AZURE_TENANT_ID="tenant id"
+```
+
+Authorize the service principal to perform key operations in your Key Vault:
+```Bash
+az keyvault set-policy --name my-key-vault --spn $AZURE_CLIENT_ID --secret-permissions get set list delete backup recover restore purge
+```
+> Possible permissions:
+> - Secret management: set, backup, delete, get, list, purge, recover, restore
+
+If you have enabled role-based access control (RBAC) for Key Vault instead, you can find roles like "Key Vault Secrets Officer" in our [RBAC guide][rbac_guide].
+
+#### Create a client
+Once the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables are set, [DefaultAzureCredential][default_cred_ref] will be able to authenticate the Client.
+
+Constructing the client also requires your vault's URL, which you can get from the Azure CLI or the Azure Portal. In the Azure Portal, this URL is the vault's "DNS Name".
+
+```golang
+cred, err := azidentity.NewDefaultAzureCredential(nil)
+
+client, err := azsecrets.NewClient("https://my-key-vault.vault.azure.net/", cred, nil)
+```
+
+## Key concepts
+### Secret
+A secret consists of a secret value and its associated metadata and management information. This library handles secret values as strings, but Azure Key Vault doesn't store them as such. For more information about secrets and how Key Vault stores and manages them, see the [Key Vault documentation](https://docs.microsoft.com/azure/key-vault/general/about-keys-secrets-certificates).
+
+Client can set secret values in the vault, update secret metadata, and delete secrets, as shown in the [examples](#examples "examples") below.
+
+## Examples
+This section contains code snippets covering common tasks:
+* [Set a Secret](#set-a-secret "Set a Secret")
+* [Retrieve a Secret](#retrieve-a-secret "Retrieve a Secret")
+* [Update Secret metadata](#update-secret-metadata "Update Secret metadata")
+* [Delete a Secret](#delete-a-secret "Delete a Secret")
+* [List Secrets](#list-secrets "List Secrets")
+
+### Set a Secret
+[SetSecret](https://aka.ms/azsdk/go/keyvault) creates new secrets and changes the values of existing secrets. If no secret with the given name exists, `SetSecret` creates a new secret with that name and the given value. If the given name is in use, `SetSecret` creates a new version of that secret, with the given value.
```

Comment:
aka.ms links are temporary or permanent this time around?

Reply:
Hoping for temporary!